src/main/query_profiler.cpp

```diff
@@ -263,24 +268,36 @@ void QueryProfiler::EndPhase() {
 	}
 }
 
-OperatorProfiler::OperatorProfiler(ClientContext &context) {
-	enabled = QueryProfiler::Get(context).IsEnabled();
-	settings = ClientConfig::GetConfig(context).profiler_settings;
-}
-
-bool OperatorProfiler::SettingIsEnabled(MetricsType metric) const {
+bool SettingIsEnabled(profiler_settings_t settings, MetricsType metric) {
```

Comment:
Do you think this should be a reference to settings? I also don't understand why we need to pass it here.

Reply:
Since I removed `profiler_settings_t settings` from `OperatorProfiler`, and now only store the relevant operator metrics, I need to pass them as a variable. But your right, they should be a reference!

Reply:
> now only store the relevant operator metrics, I need to pass them as a variable

Okay, that makes sense. Then yes, just a reference works.